src/main/java/shef/servlets/PreferenceServlet.java

```diff
@@ -0,0 +1,127 @@
+      put("american", Arrays.asList("AMERICAN"));
+      put("asian", Arrays.asList("ASIAN", "CHINESE", "INDIAN", "INDONESIAN", "PAKISTANI", "BANGLADESHI", "JAPANESE", "PHILIPINO", "FILIPINO", "VIETNAMESE", "KOREAN", "THAI"));
+      put("carribean", Arrays.asList("CARRIBEAN", "CUBAN", "HAITIAN", "DOMINICAN", "PUERTO RICAN", "JAMAICAN", "TRINIDADIAN", "COSTA RICAN"));
+      put("eastern-european", Arrays.asList("EASTERN EUROPEAN", "RUSSIAN", "UKRANIAN", "BELARUSIAN", "SLAVIC", "POLISH", "ROMANIAN", "MOLDOVAN", "SERBIAN", "SLOVAKIAN", "CROATIAN"));
+      put("western-european", Arrays.asList("WESTERN EUROPEAN", "GERMAN", "ENGLISH", "IRISH", "FRENCH", "SWISS", "BELGIAN", "DUTCH"));
+      put("latin-american", Arrays.asList("LATIN", "LATIN AMERICAN", "MEXICAN", "BRAZILIAN", "COLOMBIAN", "ARGENTINIAN", "PERUVIAN", "VENEZUELAN", "CHILEAN", "ECUADORIAN", "BOLIVIAN"));
+      put("mediterranean", Arrays.asList("MEDITERRANEAN", "GREEK", "ITALIAN", "SARDINIAN", "SPANISH"));
+      put("middle-eastern", Arrays.asList("MIDDLE EASTERN", "MIDDLE-EASTERN", "IRANIAN", "EGYPTIAN", "TURKISH", "IRAQI", "SAUDI ARABIAN", "ISRAELI", "JORDINIAN", "PALESTINIAN"));
+      put("south-asian", Arrays.asList("SOUTH ASIAN", "THAI", "CAMBODIAN", "MYANMA", "BURMESE", "INDONESIAN", "LAO", "LAOTIAN", "INDIAN"));
+
+      // Question 4: What are your general tastes?
+      put("spicy", Arrays.asList("SPICY"));
+      put("sweet", Arrays.asList("SWEET"));
+      put("salty", Arrays.asList("SALTY"));
+      put("crispy", Arrays.asList("CRISPY"));
+	  }
+  };
+
+  // Set the user's preferences
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    UserService userService = UserServiceFactory.getUserService();
+
+    // Retrieve the values of all checkboxes that the user did check in the quiz.
+    List<String> quizResponses = Arrays.asList(request.getParameterValues("choice"));
+    LinkedList<String> preferences = new LinkedList<String>();
+
+    // Add the appropriate preferences for every checkbox the user checked
+    for(String qr : quizResponses) {
+      preferences.addAll(preferenceMap.get(qr));
+    }
+
+    // Get the current user's key.
+    Key userKey = KeyFactory.createKey("User", userService.getCurrentUser().getUserId());
+    Entity user;
+
+    try {
+      // Retrieve the user from Datastore and set preferences.
+      user = datastore.get(userKey);
+      user.setProperty("preferences", preferences);
+    } catch(EntityNotFoundException e) {
+      /** This means the current user, for whom we're trying to set preferences for, doesn't exist.
+        * This should never happen. If it does, multiple things have gone seriously wrong.
+        */
+      throw new IOException("Entity not found.");
```

Comment:
Why does this throw an IOException? The error is that there's no entity, so wouldn't an EntityNotFoundException be more appropriate? I think it might make more sense to just print the stack trace and return, instead of catching an exception to throw another exception.